Sources/XcodeGenKit/PBXProjGenerator.swift

```diff
@@ -566,6 +567,45 @@ public class PBXProjGenerator {
         let childGroups = group.children.compactMap { $0 as? PBXGroup }
         childGroups.forEach(sortGroups)
     }
+    
+    func setupGroupOrdering(group: PBXGroup) {
+        let groupOrdering = project.options.groupOrdering.first { groupOrdering in
+            let groupName = group.nameOrPath
+            
+            if groupName == groupOrdering.pattern {
+                return true
+            }
+            
+            if let regex = groupOrdering.regex {
+                return regex.isMatch(to: groupName)
+            }
+            
+            return false
+        }
+        
+        if let order = groupOrdering?.order, let fileSortPosition = groupOrdering?.fileSortPosition {
+            let files = group.children.filter { $0 is PBXFileReference }
+            var groups = group.children.filter { $0 is PBXGroup }
+            
+            var filteredGroups = [PBXFileElement]()
+            
+            for groupName in order {
+                guard let group = groups.first(where: { $0.nameOrPath == groupName }) else {
```

Comment:
If this worked on just `group.children` it would be possible to include file names as well

Reply:
If we try to do the ordering for files as well, we'll face the intersection of `groupOrdering` and `groupSortPossition` options.

For example, we have the following directory:
```
MyFile1(File)
MyFile2(File)
Sources(Group)
Resources(Group)
Tests(Group)
Support files(Group)
Configurations(Group)
```
If I set up options the following way:
```
groupSortPossition: top
groupOrdering:
    - order: [MyFile1, Sources, Support files]
```
Do we expect to see the following result?
```
MyFile2(File)
MyFile1(File)
Sources(Group)
Support files(Group)
Resources(Group)
Tests(Group)
Configurations(Group)
```
And if we setup in this way?
```
groupSortPossition: bottom
groupOrdering:
    - order: [MyFile1, Sources, Support files]
```
Do we expect to see the following result?
```
MyFile1(File)
Sources(Group)
Support files(Group)
Resources(Group)
Tests(Group)
Configurations(Group)
MyFile2(File)
```

It's a bit tricky, I'd advise not to implement file ordering.

Reply:
I agree. File ordering is way less useful.